src/MessagePackAnalyzer/MessagePackCodeFixProvider.cs

```diff
@@ -118,6 +118,12 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 }
             }
 
+            if (namedSymbol.IsRecord && namedSymbol.Constructors.Any(c => IsPrimaryConstructor(c, namedSymbol)))
+            {
+                // We can`t specify the target of the attribute ([property:Key(X)] with the current roslyn version of this code fix.
```

Comment:
Would you be interested in filling this in if we can upgrade to a later version of the roslyn APIs?

Reply:
Sure

Reply:
I did some checking. I don't think the roslyn version has to do with this. At least, 4.4.0 doesn't seem to change the level of support such that `SyntaxGenerator` would let you do this. But you can do this with `SyntaxFactory` with the version we're already using. I'll see if I can address this and push an update.

Reply:
I pushed a fix.